seat/stressor_receptor_calc.py

```diff
@@ -411,103 +396,110 @@ def select_and_load_in(self) -> None:
+                "velocity threshold file": self.dlg.velocity_threshold_file.text(),
+                "velocity area of interest file": self.dlg.velocity_aoi_file.text(),
+                "paracousti device present filepath": self.dlg.paracousti_device_present.text(),
+                "paracousti device not present filepath": self.dlg.paracousti_device_not_present.text(),  # pylint: disable=line-too-long
+                "paracousti averaging": self.dlg.paracousti_averaging_combobox.currentText(),
+                "paracousti probabilities file": self.dlg.paracousti_probabilities_file.text(),
+                "paracousti_threshold_value": self.dlg.paracousti_threshold_value.text(),
+                "paracousti area of interest file": self.dlg.paracousti_aoi_file.text(),
+                "paracousti species filepath": self.dlg.paracousti_species_directory.text(),
+                "paracousti metric": self.dlg.paracousti_metric_selection_combobox.currentText(),
+                "paracousti weighting": self.dlg.paracousti_weighting_combobox.currentText(),
+                "paracousti_species_grid_resolution": self.dlg.paracousti_species_grid_resolution.text(),  # pylint: disable=line-too-long
+                "power files filepath": self.dlg.power_files.text(),
+                "power probabilities file": self.dlg.power_probabilities_file.text(),
+                "coordinate reference system": self.dlg.crs.text(),
+                "output style files": self.dlg.output_stylefile.text(),
+            }
+
+            config["Output"] = {"output filepath": self.dlg.output_folder.text()}
+
+            with open(filename, "w", encoding="utf-8") as configfile:
+                config.write(configfile)
         else:
-            layer = QgsProject.instance().addMapLayer(QgsRasterLayer(fpath, basename))
+            QgsMessageLog.logMessage(
+                "output file not given.", level=Qgis.MessageLevel.Warning
+            )
 
-    def style_layer(
+    def add_layer(
         self,
         fpath: str,
         stylepath: Optional[str] = None,
         root: Optional[QgsLayerTreeGroup] = None,
         group: Optional[QgsLayerTreeGroup] = None,
-    ) -> None:  # , ranges=True):
+        subgroup: Optional[QgsLayerTreeGroup] = None,
+    ) -> None:
         """Style and add the result layer to map."""
         basename = os.path.splitext(os.path.basename(fpath))[0]
-        if group is not None:
-            vlayer = QgsRasterLayer(fpath, basename)
-            QgsProject.instance().addMapLayer(vlayer)
-            root = QgsProject.instance().layerTreeRoot()
-            if stylepath is not None:
-                vlayer.loadNamedStyle(stylepath)
-                vlayer.triggerRepaint()
-                vlayer.reload()
-            layer = root.findLayer(vlayer.id())
+        vlayer = QgsRasterLayer(fpath, basename)
+        QgsProject.instance().addMapLayer(vlayer)
+        root = QgsProject.instance().layerTreeRoot()
```

Comment:
@tnelson-integral this root definition is overwriting any root passed to the function should this be:

```
        if root is None:
            root = QgsProject.instance().layerTreeRoot()
```